gallery_dl/extractor/fanbox.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.fanbox.cc/"""
+
+from .common import Extractor, Message
+from .. import text
+
+
+class FanboxExtractor(Extractor):
+    """Base class for fanbox extractors"""
+    category = "fanbox"
+    root = "https://www.fanbox.cc"
+    directory_fmt = ("{category}", "{creator_id}")
+    filename_fmt = "{id}_{num}.{extension}"
+    archive_fmt = "{id}_{num}"
+    _warning = True
+
+    def items(self):
+        yield Message.Version, 1
+
+        if self._warning:
+            if "FANBOXSESSID" not in self.session.cookies:
+                self.log.warning("no 'FANBOXSESSID' cookie set")
+            FanboxExtractor._warning = False
+
+        for url, data in self.posts():
+            yield Message.Directory, data
+            yield Message.Url, url, data
```

Comment:
not sure this is ok with yielding both dir and url for all posts, but i dont know how else to do it

Reply:
It is OK, but it would be better to use one `Message.Directory` for each post and a `Message.Url` for each file in that post.

Reply:
If possible, this should be rewritten so that `self.posts()` yields actual post objects (`pbody` from line 53).
Those would be used in one `yield Message.Directory, post` each, and further down you'd extract the actual files and download them with `yield Message.Url, file_url, post`.
Basically `_get_post_data_and_urls()` should be split into two parts.